### DISCUSSION TITLE
How to suppress PyInstaller errors?

### DISCUSSION THREAD

**Discussion by kr1tx** [pyinstaller/pyinstaller#7889] [Help]
```
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen Program>", line 5, in <module>
RuntimeError: protection exception (16777921)
[14524] Failed to execute script 'Program' due to unhandled exception!
```

**Comment by rokm** [comment]
That's a PyArmor exception, isn't it? You should ask them...

**Comment by kr1tx** [comment]
How? am I currently not asking them?

**Reply by rokm** [reply]
We have nothing to do with PyArmor?

**Reply by kr1tx** [reply]
Aha, I need to disable the PyInstaller's RuntimeError, I don't want that error message to appear if something went wrong. forgive me for not being clear.

**Reply by rokm** [reply]
Then you need to catch and handle the exception and suppress it (e.g., exit the application). Otherwise, PyInstaller will print unhandled exception information to console (or in Windows noconsole/windowed builds, open an unhandled exception dialog).

I suppose if your entry-point is also obfuscated/protected by PyArmor, you should create an unprotected entry-point script, and import the protected stuff inside try/except block.

**Reply by rokm** [reply]
And just to clarify, that's not "PyInstalle'rs RuntimeError"; that is PyArmor raising the exception for whatever reason, and the unhandled exception is interrupting your PyInstaller-frozen application. Same as if exception had happened while running an unfrozen python script.

**Reply by kr1tx** [reply]
I already have handled every exception in my code, still that RuntimeError raises

**Reply by rokm** [reply]
But are you handling the exceptions that PyArmor raises when it fails load your protected modules (as you try to import them)?

**Reply by kr1tx** [reply]
Yes, I gracefully have handled all pyarmor exceptions and read all the documentation and asked pyarmor's author for help even.

**Reply by rokm** [reply]
I'm out of ideas, then. As far as PyInstaller is concerned, the exception was not handled anywhere in the python code, which causes [this PyEval_EvalCode call](https://github.com/pyinstaller/pyinstaller/blob/d1b6b520a017578a19e1cb9514752a4517755ee0/bootloader/src/pyi_launch.c#L522) to return an error.

I suppose it could be that exception originates somewhere out of your control (I never looked into the details of PyArmor), but we cannot do anything about that, either. You should figure out where exception originates from (based on `protection exception (16777921)` I still think this is a PyArmor issue); but if you really want to just suppress the error message without figuring out how to handle the exception, you will need to adjust [the error handling code in  bootloader](https://github.com/pyinstaller/pyinstaller/blob/d1b6b520a017578a19e1cb9514752a4517755ee0/bootloader/src/pyi_launch.c#L527-L595), rebuild the bootloader, and rebuild your application to use the modified bootloader.

**Reply by bwoodsend** [reply]
It's possible that the exception is being raised in the pyarmor wrapper script so it's not actually touching your code.
